dlt/common/schema/schema.py

```diff
@@ -737,6 +737,26 @@ def to_pretty_yaml(
         )
         return utils.to_pretty_yaml(d)
 
+    def to_dbml(
+        self,
+        include_dlt_tables: bool = True,
+        include_internal_dlt_ref: bool = True,
+        include_parent_child_ref: bool = True,
+        include_root_child_ref: bool = True,
+        group_by_resource: bool = False,
+    ) -> str:
+        from dlt.helpers.dbml import schema_to_dbml
+
+        dbml_schema = schema_to_dbml(
```

Comment:
```py
d = self.to_dict(
            remove_defaults=remove_defaults, remove_processing_hints=remove_processing_hints
        )
```
is a crucial part of save process. it makes sure
- that schema version and hash are bumped and synced
- that various default values and Nones are removed
- that non-serializable and transient part of the schema are removed

`remove_defaults: bool = True, remove_processing_hints: bool = False` are part of the standard interface. they are used by schema storage ie. to export schemas to `export` folder (processing hints will be removed)

tldr;> make your convert function to work on `TStoredSchema`, not on `Schema` to get clean and serializable representation of `Schema`